react/features/shared-video/functions.ts

```diff
@@ -70,20 +76,71 @@ export function extractYoutubeIdOrURL(input: string) {
         return;
     }
 
-    const youtubeId = getYoutubeId(trimmedLink);
+    if (areYoutubeURLsAllowedForSharedVideo()) {
+        const youtubeId = getYoutubeId(trimmedLink);
 
-    if (youtubeId) {
-        return youtubeId;
+        if (youtubeId) {
+            return youtubeId;
+        }
     }
 
     // Check if the URL is valid, native may crash otherwise.
     try {
         // eslint-disable-next-line no-new
-        new URL(trimmedLink);
+        const url = new URL(trimmedLink);
+
+        if (!url?.hostname || !URL_WHITELIST.includes(url?.hostname)) {
```

Comment:
Isn't the first check redundant?

Reply:
not really:

![image](https://github.com/user-attachments/assets/26c0631d-607b-4bb4-9e26-062e4bd88e27)

Reply:
Actually you are correct.